ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/ipc/JsonRpcIpcServiceTest.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright contributors to Hyperledger Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.ipc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.execution.BaseJsonRpcProcessor;
+import org.hyperledger.besu.ethereum.api.jsonrpc.execution.JsonRpcExecutor;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.JsonRpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.Request;
+import org.web3j.protocol.core.methods.response.EthBlockNumber;
+import org.web3j.protocol.ipc.UnixIpcService;
+
+@EnabledOnOs({OS.LINUX, OS.MAC})
+@ExtendWith(VertxExtension.class)
+class JsonRpcIpcServiceTest {
+
+  @TempDir private Path tempDir;
+  private Vertx vertx;
+  private VertxTestContext testContext;
+
+  @BeforeEach
+  public void setUp() {
+    vertx = Vertx.vertx(new VertxOptions().setPreferNativeTransport(true));
+    testContext = new VertxTestContext();
+  }
+
+  @AfterEach
+  public void after() throws Throwable {
+    assertThat(testContext.awaitCompletion(5, TimeUnit.SECONDS))
+        .describedAs("Test completed on time")
+        .isTrue();
+    if (testContext.failed()) {
+      throw testContext.causeOfFailure();
+    }
+  }
+
+  @Test
+  void worksWithWeb3j() {
+    final Path socketPath = tempDir.resolve("besu-test.ipc");
```

Comment:
I think this test should be done in the acceptance tests with everything wired without mocks. It means that we isolate the dependency on web3j.

Reply:
Sure. I just did that. I removed this test and I added an acceptance one just checking a proper `net_version` response. I also removed the `org.web3j:core` _testDependency_ that I had added to the `:ethereum:api` module so the usage of this dependency is still isolated in the _acceptance tests_ scope.